elastic/src/main/java/org/openmbee/sdvc/elastic/NodeElasticDAOImpl.java

```diff
@@ -79,127 +84,76 @@ public boolean existsById(String indexId) {
-            // Query
-            QueryBuilder query = QueryBuilders.boolQuery()
-                .filter(QueryBuilders.termsQuery("_commitId", refsCommitIds))
-                .filter(QueryBuilders.termQuery("id", nodeId))
-                .filter(QueryBuilders.rangeQuery("_modified").lte(timestamp));
-            searchSourceBuilder.query(query);
-            searchSourceBuilder.sort(new FieldSortBuilder("_modified").order(SortOrder.DESC));
-            searchSourceBuilder.size(1);
-            //searchSourceBuilder.size(this.resultLimit);
-            searchRequest.source(searchSourceBuilder);
-            //searchRequest.scroll(scroll);
-
-            SearchResponse searchResponse = client.search(searchRequest, RequestOptions.DEFAULT);
-            //String scrollId = searchResponse.getScrollId();
-            SearchHit[] searchHits = searchResponse.getHits().getHits();
-            // :TODO going to have to iterate through the inital results I think, have to test.
-            // TODO This only returns one element, no need for scroll
-            /*
-            while (searchHits != null && searchHits.length > 0) {
-                SearchScrollRequest scrollRequest = new SearchScrollRequest(scrollId);
-                scrollRequest.scroll(scroll);
-                searchResponse = client.scroll(scrollRequest, RequestOptions.DEFAULT);
-                scrollId = searchResponse.getScrollId();
-                searchHits = searchResponse.getHits().getHits();
-                for (SearchHit hit : searchHits) {
-                    elements.add(hit.getSourceAsMap());
+        int count = 0;
+        ElementJson elementJson = newInstance();
+        while (count < refsCommitIds.size()) {
+            try {
+                List<String> sub = refsCommitIds.subList(count, Math.min(refsCommitIds.size(), count + this.termLimit));
+                SearchRequest searchRequest = new SearchRequest(getIndex());
+                SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+                // Query
+                QueryBuilder query = QueryBuilders.boolQuery()
+                    .filter(QueryBuilders
+                        .termsQuery(ElementJson.COMMITID, sub))
+                    .filter(QueryBuilders.termQuery(ElementJson.ID, nodeId))
+                    .filter(QueryBuilders.rangeQuery(ElementJson.MODIFIED).lte(timestamp));
+                searchSourceBuilder.query(query);
+                searchSourceBuilder.sort(new FieldSortBuilder("_modified").order(SortOrder.DESC));
+                searchSourceBuilder.size(1);
+                searchRequest.source(searchSourceBuilder);
+
+                SearchResponse searchResponse = client.search(searchRequest, RequestOptions.DEFAULT);
+                SearchHit[] searchHits = searchResponse.getHits().getHits();
+                if (searchHits != null && searchHits.length > 0) {
+                    elementJson.putAll(searchHits[0].getSourceAsMap());
```

Comment:
once one is found can immediately return it, otherwise hits on next loop will overwrite

Reply:
Done